### ISSUE TITLE
SULayerGetColor() and SULayerSetColor()

### ISSUE THREAD

**Issue by DanRathbun** [SketchUp/api-issue-tracker#418]
### SketchUp C API Feature Parity Request

A. Please implement `SULayerGetColor()` and `SULayerSetColor()` for the C API so as to bring it into parity with the Ruby API.

B. Perhaps deprecate `SULayerGetMaterial()` and/or add documentation notes to the effect of Thomas' comments ...
> https://forums.sketchup.com/t/feature-request-sulayersetmaterial/51877/3
> _Layer materials cannot be shared in SketchUp. The layer itself owns the material. This is different from the materials applied to entities (which is owned by the Materials collection (In Ruby this is exposed as model.materials where as in the C API you ask the ModelRef directly (I wish there had been a MaterialsRef collection in the C API as well, alas.))_
>
> _So you don’t want to mix layer materials and materials from DrawingElements - their ownership works differently._
>
> _In the C API there is no SULayerSetMaterial, but SULayerGetMaterial gets you a reference to the MaterialRef for that layer and let you modify that._

... and ...
> https://forums.sketchup.com/t/feature-request-sulayersetmaterial/51877/7
> _Internally layers own their own materials (unrelated to the model’s material collection.) Now, for some reason the C API exposed the material. (Why is not known to me - it was before me. My guess is that the original implementer wasn’t aware of this detail and simply exposed the material instead of just the colors.) So that’s some internal implementation details that unfortunately got exposed to the API. Some extensions have actually made use of that, so we are kind of stuck with it._

----

Publicly requested by Tim Hemmelman of Robert McNeil & Assoc., Rhino Development.

Reference public forum discussion:
https://forums.sketchup.com/t/feature-request-sulayersetmaterial/51877/9

**Comment by thomthom** [comment]
Feedback is that people want `layer.material` in Ruby. If we can work out the life-time challenges it's more likely that we'll add `layer.material`. But we don't add `SULayerGetColor` since you can get the color via `SULayerGetMaterial`.

Leaving this as a documentation issue.

**Comment by thomthom** [comment]
> Ah... so the current layer.color and layer.color= are just aliases into the layer's material object ?

Yes. When it was added to the Ruby API it was said that the layer material thing was an unintended consequence of how it had been implemented. Not sure why the material was given in the C API, think that was before my time. But - it's in use both in the UI and the C API and people have valid use-cases for it.

> If parity still is a goal, then SULayerGetColor could also be an shortcut to return the same result as SUMaterialGetColor, etc.

We focus on feature-parity, allowing capabilities to access SketchUp features. Not necessarily that the API should be exactly the same. Since C API expose the material it already has access to the color feature. It's more that we just have to bit the bullet and solve the life-time issues in the Ruby API and expose it there as well. It will mean the `layer.color` methods are redundant - but it is what it is. We could label it deprecated (with no intention of removing it)  but I'm not sure it's worth it, the method doesn't misbehave.
